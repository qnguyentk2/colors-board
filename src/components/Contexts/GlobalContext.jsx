import React, { createContext, useState } from 'react';
import PropTypes from 'prop-types';

const GlobalContext = createContext({});

export const GlobalProvider = ({ children }) => {
  const [isLightMode, setMode] = useState(true);
  const [isBurgerMenuOpen, setBurgerMenu] = useState(false);

  const handleChangeThemeMode = () => {
    setMode(!isLightMode);
  }

  const handleDetectBurgerMenu = (state) => {
    setBurgerMenu(state.isOpen);
  }

  const dataStore = {
    isLightMode,
    handleChangeThemeMode,
    isBurgerMenuOpen,
    handleDetectBurgerMenu
  }

  return (
    <GlobalContext.Provider value={dataStore}>
      {children}
    </GlobalContext.Provider>
  )
}

export const GlobalConsumer = GlobalContext.Consumer
export default GlobalContext;

GlobalProvider.propTypes = {
  children: PropTypes.objectOf(PropTypes.any),
};

GlobalProvider.defaultProps = {
  children: {},
};