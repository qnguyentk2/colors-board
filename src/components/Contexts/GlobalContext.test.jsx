/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { GlobalProvider } from './GlobalContext';

const setup = () => {
  const wrapper = shallow(<GlobalProvider />);

  return { wrapper };
}

describe('Test GlobalProvider', () => {
  test('renders GlobalProvider without crashing', () => {
    const { wrapper } = setup();
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
})