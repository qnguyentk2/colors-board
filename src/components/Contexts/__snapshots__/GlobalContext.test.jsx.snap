// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GlobalProvider renders GlobalProvider without crashing 1`] = `
<ContextProvider
  value={
    Object {
      "handleChangeThemeMode": [Function],
      "handleDetectBurgerMenu": [Function],
      "isBurgerMenuOpen": false,
      "isLightMode": true,
    }
  }
>
  <Component />
</ContextProvider>
`;
